Elasticsearch/ElasticsearchClient.php

```diff
@@ -115,12 +120,13 @@ public function sendRequests (array $requests)
      */
     public function sendBulkIndexRequests (array $requests)
     {
+        /** @var IndexDocumentRequest[] $requests */
+        $requests = \array_values($requests);
         $currentBulk = [];
         $maxIndex = count($requests) - 1;
 
-        for ($i = 0; $i <= $maxIndex; $i++)
+        foreach ($requests as $i => $request)
         {
-            $request = $requests[$i];
             $data = $request->getData();
 
             // add header
```

Comment:
The `$this->client->bulk()` call in line 147 is still not protected from throwing exceptions.